### ISSUE TITLE
lr_scheduler was disabled after warmup_steps

### ISSUE THREAD

**Issue by kelvinqin** [sooftware/kospeech#129]
## Title
- lr_scheduler was disabled after warmup_steps (400 for example)

## Description
in bin/main.py,
optimizer = Optimizer(optimizer, lr_scheduler, config.train.warmup_steps, config.train.max_grad_norm)

Here, not sure why you assign scheduel_period as config.warmup_steps? Which cause lr_scheduler disabled after warmup_steps:

if self.scheduler_period == self.count:
    self.scheduler = None
    self.scheduler_period = 0
    self.count = 0

## Linked Issues
  
- resolved #

**Comment by sooftware** [comment]
I fix this issue.

**Comment by kelvinqin** [comment]
Soohwan,
Thanks, but that introduce anther bug, config.train.total_steps was neither initialized or defined in the default config, should it be as following?
        #optimizer = Optimizer(optimizer, lr_scheduler, config.train.total_steps, config.train.max_grad_norm)
        optimizer = Optimizer(optimizer, lr_scheduler, int(config.train.num_epochs * epoch_time_step), config.train.max_grad_norm)

**Comment by sooftware** [comment]
I define `total_steps` in `kospeech.trainer.__init__.TrainConfig`. But, you can set by `optimizer = Optimizer(optimizer, lr_scheduler, int(config.train.num_epochs * epoch_time_step), config.train.max_grad_norm)`.

**Comment by kelvinqin** [comment]
Soohwan,
Thanks to let me know this and thanks for ur work, I am building a Mandarin ASR using deepspeech2 framework of yours, will share with you once I got a satisfied result.

The current result is not good, I am not sure if it was caused by the incorrect "disable of lr_scheduler", will try after fix this. (The current CER is abnormally high, which is 34%, it should be close to 6.x% using CTC approach in my previous work)
Kelvin

**Comment by sooftware** [comment]
Check the configuration. And there may be an error because it is implemented as I understand it. Feel free to give me feedback if you have a problem. Thank you.

**Comment by kelvinqin** [comment]
Ok, thanks, keep touch
